alembic/versions/614911daf883_add_seasonal_to_time_resolution_enum.py

```diff
@@ -0,0 +1,145 @@
+#
+# TESTING:
+#
+# These migrations have been tested on a SQLite test database and on
+# a full copy of the production database. Both upgrade and downgrade work
+# on both databases.
+
+old_options = (
+    '1-minute', '2-minute', '5-minute', '15-minute', '30-minute',
+    '1-hourly', '3-hourly', '6-hourly', '12-hourly', 'daily',
+    'monthly', 'yearly', 'other', 'irregular',
+)
+new_options = (
+    '1-minute', '2-minute', '5-minute', '15-minute', '30-minute',
+    '1-hourly', '3-hourly', '6-hourly', '12-hourly', 'daily',
+    'monthly', 'seasonal', 'yearly', 'other', 'irregular',
+)
+
+
+def alter_column(curr_options, dest_options):
+    """
+    Alter columns, which in SQLite alters the check constraint. Yay.
+    See http://alembic.zzzcomputing.com/en/latest/batch.html
+
+    :param curr_options: tuple of options (members) in the current enum type
+    :param dest_options: tuple of options (members) in the destination enum type
+    :return: None
+    """
+    old_type = sa.Enum(*curr_options, name='timescale')
+    new_type = sa.Enum(*dest_options, name='timescale')
+    with op.batch_alter_table('time_sets') as batch_op:
+        batch_op.alter_column(
+            'time_resolution', type_=new_type, existing_type=old_type)
+
+
+def swap_and_drop(curr_options, dest_options):
+    """
+    Create a temporary enum type with the destination options in it,
+    swap it for the current enum type with the current options in it,
+    then rename the temporary and drop the temporary and old enum types.
+    This works for PostgreSQL.
+
+    Adapted from https://stackoverflow.com/questions/14845203/altering-an-enum-field-using-alembic
+
+    :param curr_options: tuple of options (members) in the current enum type
+    :param dest_options: tuple of options (members) in the destination enum type
+    :return: None
+    """
+    old_type = sa.Enum(*curr_options, name='timescale')
+    new_type = sa.Enum(*dest_options, name='timescale')
+    tmp_type = sa.Enum(*dest_options, name='_timescale')
+    # Create a temporary enum type, convert to it and drop the old enum type
+    tmp_type.create(connection, checkfirst=False)
+    op.execute('ALTER TABLE time_sets '
+               'ALTER COLUMN time_resolution TYPE _timescale '
+               'USING time_resolution::text::_timescale')
+    old_type.drop(connection, checkfirst=False)
+    # Create and convert to the new enum type
+    new_type.create(connection, checkfirst=False)
+    op.execute('ALTER TABLE time_sets '
+               'ALTER COLUMN time_resolution TYPE timescale '
+               'USING time_resolution::text::timescale')
+    # Drop the temporary enum type
+    tmp_type.drop(connection, checkfirst=False)
+
+
+def upgrade():
+    if dialect == 'sqlite':
+        alter_column(old_options, new_options)
+    elif dialect == 'postgresql':
+        swap_and_drop(old_options, new_options)
+    else:
+        raise RuntimeError('This migration is not implemented for dialect {}'
```

Comment:
Maybe it'd be better to warn that the migration is not *tested* for this dialect? Most backends *will* have `ALTER TABLE`, presumably. Maybe we should be generous?